File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/response/CartResponse.java
package com.coffee.gu.api.controller.v1.response;

import com.coffee.gu.cart.CartItem;

import java.math.BigDecimal;
import java.util.List;

public record CartResponse(
        List<CartItemResponse> items
) {


    public record CartItemResponse(
            Long id,
            Long menuId,
            String menuName,
            String imageUrl,
            String description,
            BigDecimal costPrice,
            BigDecimal salesPrice,
            Long quantity) {
        public static CartItemResponse from(CartItem cartItem) {
            return new CartItemResponse(
                    cartItem.getId(),
                    cartItem.getMenu().getId(),
                    cartItem.getMenu().getName(),
                    cartItem.getMenu().getImageUrl(),
                    cartItem.getMenu().getDescription(),
                    cartItem.getMenu().getCostPrice(),
                    cartItem.getMenu().getSalesPrice(),
                    cartItem.getQuantity()
            );
        }
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/response/CartResponse.kt
package com.coffee.gu.api.controller.v1.response

import com.coffee.gu.cart.CartItem
import java.math.BigDecimal

class CartResponse(
    val items: List<CartItemResponse>
) {
    class CartItemResponse(
        val id: Long,
        val menuId: Long?,
        val menuName: String?,
        val imageUrl: String?,
        val description: String?,
        val costPrice: BigDecimal?,
        val salesPrice: BigDecimal?,
        val quantity: Long,
    ) {
        companion object {
            @JvmStatic
            fun from(cartItem: CartItem): CartItemResponse {
                val menu = cartItem.menu
                return CartItemResponse(
                    id = cartItem.id,
                    menuId = menu?.id,
                    menuName = menu?.name,
                    imageUrl = menu?.imageUrl,
                    description = menu?.description,
                    costPrice = menu?.costPrice,
                    salesPrice = menu?.salesPrice,
                    quantity = cartItem.quantity
                )
            }
        }
    }
}
